.ci/scripts/get_versions.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+"""
+This script fetches the available versions of Elastic Agent from the Elastic website and
+performs various operations on the versions.
+
+Functions:
+- get_available_versions:
+    Fetches the available versions of Elastic Agent.
+- parse_version:
+    Parses a version string into a tuple of integers.
+- filter_versions:
+    Filters a list of versions based on the given prefix or after a specific version.
+- get_package_version:
+    Retrieves the package version of cloud_security_posture for a given Kibana version.
+- generate_job_matrix:
+    Generates a job matrix based on the given versions.
+- main:
+    Retrieves available versions of Elastic Agent and performs operations on them.
+"""
+import os
+import json
+import argparse
+import requests
+
+
+def get_available_versions():
+    """
+    Fetches the available versions of Elastic Agent from the Elastic API.
+
+    Returns:
+        list: A list of the latest versions of Elastic Agent.
+
+    Raises:
+        Exception: If there is an error while fetching the versions list.
+    """
+    url = "https://www.elastic.co/api/product_versions"
+    headers = {"Content-Type": "application/json"}
+
+    try:
+        response = requests.get(url, headers=headers, timeout=60)
+        response.raise_for_status()
+        json_body = response.json()
+
+        versions_dict = {}
+        for item in json_body[0]:
+            title = item.get("title", "")
+            version_number = item.get("version_number", "")
+            # Filter out versions with no alphabetic characters in the version number
+            if "Elastic Agent" in title and not any(char.isalpha() for char in version_number):
+                major_minor = ".".join(version_number.split(".")[:2])
+                versions_dict[major_minor] = max(versions_dict.get(major_minor, ""), version_number)
+
+        # Sort versions and select the latest num_versions
+        versions = sorted(
+            list(versions_dict.values()),
+            key=lambda x: tuple(map(int, x.split("."))),
+            reverse=True,
+        )
+
+        return versions
+    except requests.exceptions.RequestException as e:
+        raise requests.exceptions.RequestException("Failed to fetch versions list") from e
+
+
+def parse_version(version):
+    """
+    Parses a version string into a tuple of integers.
+
+    Args:
+        version (str): The version string to parse.
+
+    Returns:
+        tuple: A tuple of integers representing the parsed version.
+
+    Example:
+        >>> parse_version('1.2.3')
+        (1, 2, 3)
+    """
+    return tuple(map(int, version.split(".")))
+
+
+def filter_versions(versions, prefix=None, after=None):
```

Comment:
nit: I think this naming could be a bit clearer
```suggestion
def filter_versions(versions, prefix=None, min_version=None):
```

Reply:
Renaming the argument wasn't enough, so I updated the whole function.